### PR TITLE
Allow setting “no draw on zoom flag” in image JSON

### PR DESCRIPTION
**Pull request by spacek531**
There are several flags that could be exposed to json for convenience of custom object creators. This is one of them.

### CODE REVIEW

File: src/openrct2/drawing/ImageImporter.cpp [hunk 1]
```diff
@@ -388,17 +390,23 @@ namespace OpenRCT2::Drawing
         auto yOffset = Json::GetNumber<int16_t>(input["y"]);
         auto keepPalette = Json::GetString(input["palette"]) == "keep";
         auto palette = keepPalette ? Palette::KeepIndices : Palette::OpenRCT2;
-
+        auto flags = EnumValue(ImportFlags::None);
         auto raw = Json::GetString(input["format"]) == "raw";
-        auto flags = raw ? ImportFlags::None : ImportFlags::RLE;
-
+        if (raw)
+            flags |= EnumValue(ImportFlags::RLE);
```
Comment (AaronVanGeffen) [inline src/openrct2/drawing/ImageImporter.cpp:396]: Could you add some braces around this for legibility? i.e.
```suggestion
        if (raw)
        {
            flags |= EnumValue(ImportFlags::RLE);
        }
```

Comment (Gymnasiast) [reply src/openrct2/drawing/ImageImporter.cpp:396]: We generally don’t do that. I do think a blank line before and after is a good thing, though.

**Comment by Gymnasiast** [comment]
> There are several flags that could be exposed to json for convenience of custom object creators. This is one of them.

What are they? I’d prefer doing this right - once we expose something to JSON, we can’t really remove it, so I’d like to get this right first time.

**Comment by spacek531** [comment]
> > There are several flags that could be exposed to json for convenience of custom object creators. This is one of them.
> 
> What are they? I’d prefer doing this right - once we expose something to JSON, we can’t really remove it, so I’d like to get this right first time.

It turns out I misspoke. I saw there were other flags in the G1_FLAG enum but none of them are useful aside from zoom sprite - which already has representation in the json, although it appears the game ignores it twice.

**Comment by spacek531** [comment]
I don't know why tests are failing

File: src/openrct2/drawing/ImageImporter.cpp [hunk 2]
```diff
@@ -388,17 +390,25 @@ namespace OpenRCT2::Drawing
         auto yOffset = Json::GetNumber<int16_t>(input["y"]);
         auto keepPalette = Json::GetString(input["palette"]) == "keep";
         auto palette = keepPalette ? Palette::KeepIndices : Palette::OpenRCT2;
+        auto flags = EnumValue(ImportFlags::None);
 
         auto raw = Json::GetString(input["format"]) == "raw";
-        auto flags = raw ? ImportFlags::None : ImportFlags::RLE;
+        if (raw)
+            flags |= EnumValue(ImportFlags::RLE);
```
Comment (Gymnasiast) [inline src/openrct2/drawing/ImageImporter.cpp:397]: You reversed what the if does here.

**Comment by Gymnasiast** [comment]
How is the zoom sprite "broken"?

**Comment by spacek531** [comment]
There's a flag that needs to be set for the image list to accept zoom sprites and not use the original. That flag is only set when loading G1 - there are no codepaths that set it when loading from objects of any image storage mechanism. It's been a while since I looked at this code and I have to refamiliarize myself to give a more detailed answer.

**Comment by spacek531** [comment]
Tests pass by fixing the reversed case, so I've opened this PR for review.

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**